app.rb

```diff
@@ -0,0 +1,57 @@
+require './person'
+require './book'
+require './rental'
+require './teacher'
+require './student'
+
+# app file to hold all operations
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "#{book.id}: #{book.title} by #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      puts "#{person.id}: #{person.name}"
+    end
+  end
+
+  def create_person(type, id, name, parent_permission)
+    if type == 'teacher'
+      @people << Teacher.new(id, name)
+    elsif type == 'student'
+      @people << Student.new(id, name, parent_permission: parent_permission)
+    else
+      puts 'Invalid person type.'
+    end
+  end
```

Comment:
While trying to create a student, what value is expected for parent_permission? You should indicate this in your UI so users can know if it is a `Yes`, `No`, `Y`, `N`, `true` or `false`. I got the below error when I inputted `Y` as expected. 

![image](https://github.com/Daniel-Kigozi/school_library/assets/84156714/74f48bde-d93b-49f8-83ab-3b292d7c2d9a)


See my suggestion of how your create_person method would look like. 


```ruby
def create_person
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    # *If needed, ask for parameters for the option.
    person_type = gets.chomp
    if person_type != "1" && person_type != "2"
      puts "Invalid option"
      return
    end

    print "Age: "
    # *If needed, ask for parameters for the option.
    age = gets.chomp

    print "Name: "
    # *If needed, ask for parameters for the option.
    name = gets.chomp

    person =
      if person_type == "1"
        print "Has parent permission? [Y/N]: "
        # *If needed, ask for parameters for the option.
        parent_permission = gets.chomp
        parent_permission = parent_permission.downcase == "y"

        Student.new(age, name, parent_permission)
      elsif person_type == "2"
        print "Specialization: "
        # *If needed, ask for parameters for the option.
        specialization = gets.chomp

        Teacher.new(age, specialization, name)
      end

    @people << person
    puts "Person created successfully"
  end
```